src/open_payments_sdk/http.py

```diff
@@ -1,30 +1,43 @@
-import logging
-
+"""
+HTTP Client 
+"""
 import httpx
 
-from open_payments_sdk import configuration
-
+class HttpClient:
+    """
+    HTTP Client
+    """
+    http_timeout: float
 
-class Client:
-    def __init__(self, cfg: configuration.Configuration):
-        self.logger = logging.getLogger(__name__)
-        self.logger.addHandler(cfg.get_log_handler())
-        self.user_agent = cfg.user_agent
+    def __init__(self, http_timeout: float):
+        self.http_timeout = http_timeout
 
-    @staticmethod
-    def get(url, params=None, headers=None):
-        res = httpx.get(url, params=params, headers=headers)
-        res.raise_for_status()
-        return res.text
-
-    @staticmethod
-    def post(url, json=None, headers=None):
-        res = httpx.post(url, json=json, headers=headers)
-        res.raise_for_status()
-        return res.text
+    def build_request(
+            self,
+            method: str,
+            url: str,
+            headers = None,
+            data = None,
+            json: dict = None,
+            params: dict = None
+    ) -> httpx.Request:
```

Comment:
Should we expose this `httpx` dependency like this to other modules? Seems fine from perspective of better control at call-site, but makes dependency harder to switch when/if needed.